source/create.ts

```diff
@@ -61,9 +61,11 @@ export interface GotRequestMethod {
 	<T>(url: string | Merge<Options, {isStream: true}>, options?: Merge<Options, {isStream: true}>): ProxyStream<T>;
 }
 
+export type GotPaginateOptions<T> = Except<Options, keyof PaginationOptions<unknown>> & PaginationOptions<T>;
+
 export interface GotPaginate {
-	<T>(url: URLOrOptions & PaginationOptions<T>, options?: Options & PaginationOptions<T>): AsyncIterableIterator<T>;
-	all<T>(url: URLOrOptions & PaginationOptions<T>, options?: Options & PaginationOptions<T>): Promise<T[]>;
+	<T>(url: string | GotPaginateOptions<T>, options?: GotPaginateOptions<T>): AsyncIterableIterator<T>;
```

Comment:
I think `string | GotPaginateOptions<T>` should be `URLOrOptions & GotPaginateOptions<T>`

Reply:
That would duplicate work but also duplicate the options because `GotPaginateOptions` is defined like this:
```ts
export type GotPaginateOptions<T> = Except<Options, keyof PaginationOptions<unknown>> & PaginationOptions<T>;
```

Though, I just pushed a commit which adds the `URLOrGotPaginateOptions` type so we have that definition in one place.